### PR TITLE
test: fix  TestTaskError and TestMultiError flaky tests

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve? 
close https://github.com/pingcap/ticdc/issues/2469

### What is changed and how it works?

In one case, it did happen that isCancelled was used to the next event before it was updated, and ErrCh immediately received an error, just as cancel was called during the execution of AddEvent, which is why I encountered this unstable test. I was able to reproduce the problem occasionally after increasing the number of events and limiting resources locally.

### Check List 

Tests 

 - Unit test
 

Code changes

None

Side effects

None

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by overvenus — APPROVED** [review]
LGTM, could you check if we have handle `context canceled` error properly when we call `AddEvent` in other files?

**Comment by hi-rustin** [comment]
> LGTM, could you check if we have handle `context canceled` error properly when we call `AddEvent` in other files?

@overvenus It looks like calls elsewhere will return this error directly.

**Comment by hi-rustin** [comment]
@amyangfei @ben1009  I think it ready for review.
Could you please take a look? Thanks!

**Comment by hi-rustin** [comment]
I have tested this many times by limiting resources to 1C locally and it works.

File: pkg/workerpool/pool_test.go [hunk 1]
```diff
@@ -55,7 +54,10 @@ func (s *workerPoolSuite) TestTaskError(c *check.C) {
 		c.Assert(err, check.ErrorMatches, "test error")
 	})
 
+	var wg sync.WaitGroup
+	wg.Add(1)
 	errg.Go(func() error {
```
Comment (amyangfei) [inline pkg/workerpool/pool_test.go:59]: `errg.Go` is not necessary for this goroutine

File: pkg/workerpool/pool_test.go [hunk 2]
```diff
@@ -130,7 +134,10 @@ func (s *workerPoolSuite) TestMultiError(c *check.C) {
 		return nil
 	})
 
+	var wg sync.WaitGroup
+	wg.Add(1)
 	errg.Go(func() error {
```
Comment (amyangfei) [inline pkg/workerpool/pool_test.go:139]: ditto

**Review by amyangfei — APPROVED** [review]

File: pkg/workerpool/pool_test.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/workerpool/pool_test.go]: Thanks for your review!  💚 💙 💜 💛 ❤️

**Comment by hi-rustin** [comment]
/assign @amyangfei 

Could you please help me trigger the merge process? Thanks!

**Comment by amyangfei** [comment]
needs to be cherry picked to release branches

**Pull request merged**